src/rTerminal.ts

```diff
@@ -18,7 +21,17 @@ export function createRTerm(preserveshow?: boolean): boolean {
         const termOpt: string[] = config.get("rterm.option");
         pathExists(termPath, (err, exists) => {
             if (exists) {
-                rTerm = window.createTerminal(termName, termPath, termOpt);
+                let termOptions: TerminalOptions = {
+                    name: termName,
+                    shellPath: termPath,
+                    shellArgs: termOpt
+                };
+                if (config.get("sessionWatcher")) {
+                    termOptions.env = {
+                        R_PROFILE_USER: path.join(os.homedir(), ".vscode-R", ".Rprofile")
```

Comment:
You might want to check if the user has defined a custom `R_PROFILE_USER`.
If it is not empty, you might need to define `R_PROFILE_USER_OLD` and source `R_PROFILE_USER_OLD` instead of `~/.Rprofile`

Reply:
Yes, we should respect existing `R_PROFILE_USER`.